### ISSUE TITLE
Problematic flags for the Clang compiler cause errors from warnings

### ISSUE THREAD

**Issue by paspro** [code-saturne/code_saturne#106]
For the Clang compiler to work properly (at least the latest version), one needs to modify file cs_auto_flags.sh, line 400 which reads:

`cflags_default="-funsigned-char -Wall -Wshadow -Wpointer-arith -Wmissing-prototypes -Wuninitialized -Wunused"`

and change it to:

`cflags_default="-funsigned-char -Wall -Wshadow -Wpointer-arith -Wmissing-prototypes -Wuninitialized -Wunused -Wno-incompatible-function-pointer-types -Wno-implicit-int"`

in order to avoid warnings of "incompatible function pointer types" and "implicit integer" to be marked as errors. Otherwise, one should update the source code to fix these warnings.

**Comment by YvanFournier** [comment]
Hello,

I can't reproduce this with Clang 11. I'll try on another machinec with Clang 15 (I prefer updating the source code to changing the warnings).

Note: the automatic commit reference actually references the previous issue. Sorry for the typo.

**Comment by YvanFournier** [comment]
Hello @paspro, is this issue still present ? I have not reproduced it on various clang or clang-based builds ? (with version ranging frome clang 14 to clang 22 at least).

**Comment by paspro** [comment]
Indeed, this problem is no longer present in the current / more recent versions of the compiler.
